pkg/action/bootentries.go

```diff
@@ -0,0 +1,212 @@
+	"github.com/kairos-io/kairos-agent/v2/pkg/utils/partitions"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+)
+
+func SelectBootEntry(cfg *config.Config, entry string) error {
+	// Read the systemd-boot conf file
+	if !strings.HasSuffix(entry, ".conf") {
+		entry = entry + ".conf"
+	}
+
+	cfg.Logger.Infof("Setting default boot entry to %s", entry)
+
+	// Get EFI partition
+	efiPartition, err := GetEfiPartition()
+	if err != nil {
+		return err
+	}
+	// Validate entry exists
+	entries, err := ListSystemdEntries(cfg, efiPartition)
+	if err != nil {
+		return err
+
+	}
+	// Check that entry exists in the entries list
+	found := false
+	for _, e := range entries {
+		if e == entry {
+			found = true
+			break
+		}
+	}
+	if !found {
+		cfg.Logger.Errorf("entry %s does not exist", entry)
+		cfg.Logger.Debugf("entries: %v", entries)
+		return fmt.Errorf("entry %s does not exist", entry)
+	}
+	// Get EFI partition
+	parts, err := partitions.GetAllPartitions()
+	if err != nil {
+		return fmt.Errorf("could not read host partitions: %s", err)
+	}
+	for _, p := range parts {
+		if p.FilesystemLabel == constants.EfiLabel {
+			efiPartition = p
+			break
+		}
+	}
+	if efiPartition == nil {
+		return fmt.Errorf("could not find EFI partition")
+	}
+
+	// Mount it RW
+	err = syscall.Mount("", efiPartition.MountPoint, "", syscall.MS_REMOUNT, "")
+	if err != nil {
+		cfg.Logger.Errorf("could not remount EFI partition: %s", err)
+		return err
+	}
+	// Remount it RO when finished
+	defer func(source string, target string, fstype string, flags uintptr, data string) {
+		err = syscall.Mount(source, target, fstype, flags, data)
+		if err != nil {
+			cfg.Logger.Errorf("could not remount EFI partition as RO: %s", err)
+		}
+	}("", efiPartition.MountPoint, "", syscall.MS_REMOUNT|syscall.MS_RDONLY, "")
+
+	systemdConf, err := SystemdBootConfReader(filepath.Join(efiPartition.MountPoint, "loader/loader.conf"))
+	if err != nil {
+		cfg.Logger.Errorf("could not read loader.conf: %s", err)
+		return err
+	}
+	// Set the default entry to the selected entry
+	systemdConf["default"] = entry
+	err = SystemdBootConfWriter(filepath.Join(efiPartition.MountPoint, "loader/loader.conf"), systemdConf)
+	if err != nil {
+		cfg.Logger.Errorf("could not write loader.conf: %s", err)
+		return err
+	}
+	cfg.Logger.Infof("Default boot entry set to %s", entry)
+	return err
+}
+
+func ListBootEntries(cfg *config.Config) error {
+	// Get EFI partition
+	efiPartition, err := GetEfiPartition()
+	if err != nil {
+		return err
+	}
+	// Get default entry from loader.conf
+	loaderConf, err := utils.SystemdBootConfReader(filepath.Join(efiPartition.MountPoint, "loader/loader.conf"))
+	if err != nil {
+		return err
+	}
+
+	entries, err := ListSystemdEntries(cfg, efiPartition)
+
+	// create a selector
+	selector := selection.New(fmt.Sprintf("Select Boot Entry (current entry: %s)", loaderConf["default"]), entries)
+	selector.Filter = nil        // Remove the filter
```

Comment:
I think it would be better to have a `selector.RemoveFilter()` function so we don't need to rely on comments so much